doc/source/development/extending.rst

```diff
@@ -134,6 +134,14 @@ by some other storage type, like Python lists.
 See the `extension array source`_ for the interface definition. The docstrings
 and comments contain guidance for properly implementing the interface.
 
+:class:`~pandas.api.extensions.NDArrayBackedExtensionArray`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For ExtensionArrays backed by a single NumPy array, the
+:class:`~pandas.api.extensions.NDArrayBackedExtensionArray` class can save you
+some effort. It contains a private property ``_ndarray`` with the backing NumPy
+array and implements the extension array interface.
+
```

Comment:
Could also mention that they can optionally subclass `tests.extension.base.NDArrayBacked2DTests` in the test suite

Reply:
Could call. Added to the docs, along with a minimal code sample.

I'll start using that in the implementations at https://github.com/googleapis/python-db-dtypes-pandas too. Looks like it might have already caught some issues.

Reply:
> Looks like it might have already caught some issues.

If there's any chance that the problem is with the tests, definitely let me know.  AFAIK yours is the first 3rd-party EA to use these tests, so there may be kinks to work out.

Reply:
I think it was a true failure, and a relatively easy fix, I think (WIP at https://github.com/googleapis/python-db-dtypes-pandas/pull/64)

Was missing the logic like

https://github.com/pandas-dev/pandas/blob/4f712c1c48803650e2be8a3bac9d3cfc047b3da6/pandas/core/arrays/datetimelike.py#L1529-L1531

to support 2D arrays.